simple: Add source and destination port options + cleanup

a-ilango opened the pull request:
Add an option for specifying both source and destination ports.
Pass source address info only via hints. Remove redundant code 
that did the same using fi_getinfo parameters.

If these changes are fine I'll update the other examples. I've kept the port in cs_opts so that it doesn't break other examples for now.

This should fix #130 .

@shantonu @jithinjosepkl Let me know if these look good.

shefty commented on common/shared.c:
```diff
@@ -336,6 +336,10 @@ void ft_parsecsopts(int op, char *optarg, struct cs_opts *opts)
 		break;
 	case 'p':
 		opts->port = optarg;
+		opts->src_port = optarg;
```
Let's not keep both port and src_port.  'port' can be removed.

a-ilango replied:
Yes, I'll be removing the port when I update other examples.

Sounds good. -p for dst_port and -b for src_port.

shantonu commented on simple/rdm_pingpong.c:
```diff
@@ -302,18 +302,9 @@ static int bind_ep_res(void)
-	}
-
-	ret = fi_getinfo(FT_FIVERSION, node, opts.port, flags, &hints, &fi);
+	ret = fi_getinfo(FT_FIVERSION, opts.dst_addr, opts.dst_port, 0, &hints, &fi);
```
The simple examples don't have  -s (source address) and -p option. For those, we can leave the code as it is for simplicity.

a-ilango replied:
Sure.

shantonu commented:
Can you also modify ft_csusage function in common/shared.c to add the correct description of the port option?

a-ilango commented:
>Can you also modify ft_csusage function in common/shared.c to add the correct description of the port option?

Good catch. I'll update ft_csusage.

shefty commented:
Is this pull request ready for merging?

a-ilango commented:
Not yet. I'm waiting for #147 to be merged. Since that PR changes a lot of files, it would be easier for me to modify files after it is merged.

shefty commented:
I merged #147.

a-ilango commented:
Thanks, I'll update soon.

a-ilango commented:
A slight modification in the changes. I have decided to retain passing source address information for server via fi_getinfo. Now I call ft_getsrcaddr for only the client. I have updated all the relevant files.
@shantonu @shefty Please review the changes.
@goodell I have also updated ud_pingpong. Please let me know if the changes are fine.

goodell commented on common/shared.c:
```diff
@@ -283,7 +283,8 @@ void ft_csusage(char *name, char *desc)
 	fprintf(stderr, "\nOptions:\n");
 	fprintf(stderr, "  -n <domain>\tdomain name\n");
-	fprintf(stderr, "  -p <port>\tnon default port number\n");
+	fprintf(stderr, "  -b <src_port>\tnon default source port number\n");
```
I'd prefer `-P` instead of `-b`.

a-ilango replied:
Quoting @shefty from previous comments:
>Also, I would keep -p to mean the dst_port, which is the common case. For the source port, maybe use -b? (for bind port)

Using -b looks fine to me. Let me know what you think.

goodell replied:
Missed that.  `-b` isn't a deal breaker, just not my preference.  If @shefty is good with it, feel free to ignore me.

goodell commented:
`fi_ud_pingpong` changes seem fine and work OK in my testing with usnic
